testcases/OpTestPCI.py

```diff
@@ -0,0 +1,112 @@
+    #
+    # "Only required for inband tests" else Default = None
+    # @param i_hostIP The IP address of the HOST
+    # @param i_hostuser The userid to log into the HOST
+    # @param i_hostPasswd The password of the userid to log into the HOST with
+    #
+    def __init__(self, i_bmcIP, i_bmcUser, i_bmcPasswd,
+                 i_bmcUserIpmi, i_bmcPasswdIpmi, i_ffdcDir=None, i_hostip=None,
+                 i_hostuser=None, i_hostPasswd=None, i_hostLspci=None):
+        self.cv_BMC = OpTestBMC(i_bmcIP, i_bmcUser, i_bmcPasswd, i_ffdcDir)
+        self.cv_IPMI = OpTestIPMI(i_bmcIP, i_bmcUserIpmi, i_bmcPasswdIpmi,
+                                  i_ffdcDir, i_hostip, i_hostuser, i_hostPasswd)
+        self.cv_HOST = OpTestHost(i_hostip, i_hostuser, i_hostPasswd, i_bmcIP, i_ffdcDir)
+        self.cv_SYSTEM = OpTestSystem(i_bmcIP, i_bmcUser, i_bmcPasswd,
+                                      i_bmcUserIpmi, i_bmcPasswdIpmi, i_ffdcDir, i_hostip,
+                                      i_hostuser, i_hostPasswd)
+        self.util = OpTestUtil()
+        self.lspci_file = i_hostLspci
+
+    ##
+    # @brief  This function will get the PCI and USB susbsytem Info
+    #         And also this test compares known good data of
+    #         "lspci -mm -n" which is stored in testcases/data directory
+    #         with the current lspci data.
+    #         User need to specify the corresponding file name into
+    #         machines xml like lspci.txt which contains "lspci -mm -n"
+    #         command output in a working good state of system.
+    #         tools used are lspci and lsusb
+    #
+    # @return BMC_CONST.FW_SUCCESS or BMC_CONST.FW_FAILED
+    #
+    def test_list_pci_devices_info(self):
+        self.cv_SYSTEM.sys_hard_reboot()
+        self.cv_HOST.host_check_command("lspci")
+        self.cv_HOST.host_check_command("lsusb")
+        self.cv_HOST.host_list_pci_devices()
+        self.cv_HOST.host_get_pci_verbose_info()
+        self.cv_HOST.host_list_usb_devices()
+        if self.lspci_file == "empty.txt":
+            print "Skipping the pci devices comparision as missing the lspci data file name in machines xml"
+            return BMC_CONST.FW_SUCCESS
+        else:
+            filename = os.path.join(os.path.dirname(__file__), "data/"+self.lspci_file)
+            if not os.path.isfile(filename):
+                raise OpTestError("lspci file %s not found in data directory" % filename)
+            with open(filename, 'r') as f:
+                l_good_data = f.read().replace('\n', '')
+            print l_good_data
+            l_res = self.cv_HOST.host_run_command("lspci -mm -n")
+            l_lspci_data = l_res.replace("\r\n", "")
+            if l_good_data == l_lspci_data:
+                print "All the pci devices are detected by firmware"
+            else:
+                print l_lspci_data
+                raise OpTestError("There is a mismatch b/w known good output and tested lspci output")
```

Comment:
It'd be good to call out to diff to print the actual difference between expected/detected.